### DISCUSSION TITLE
morphological feature not working (TypeError)

### DISCUSSION THREAD

**Discussion by nandinib1999** [explosion/spaCy#8528] [Help: Other Questions] [answered]
## How to reproduce the behaviour
doc = en("python execution.")
for token in doc:
  print(token.text, "->", token.pos_, '->', token.dep_)
  print(token.morph)

![image](https://user-images.githubusercontent.com/41077745/123627700-f0cd9700-d82f-11eb-8f5c-c4373f5aa555.png)

## Your Environment

* Python Version Used: 3.7.10
* spaCy Version Used: 2.2.4
* Environment Information: Google Colab

**Comment by adrianeboyd** [comment, marked as answer]
I'm not sure where this code is from, but `token.morph` was only partially implemented in spacy v2, so this isn't going to work with v2.2.4.

If you want to use it, it'd be easiest to upgrade to spacy v3 (`pip install -U spacy`). You'll need to upgrade the downloaded `en_core_web_md` model, too (`spacy download en_core_web_md`).

**Reply by nandinib1999** [reply]
Thank you for your quick response. I didn't realize that "token.morph" was only available for spacy v3. It worked smoothly after upgrading the spacy.
